flask/packages.py

```diff
@@ -114,3 +114,32 @@ def upload():
 
         return jsonify({"message": "Package Uploaded Successfully.", "code": 200})
 
+@app.route("/packages/<username>/<namespace_name>/<package_name>/delete", methods=["GET", "POST"])
+def delete_package(username, namespace_name, package_name):
+    uuid = request.cookies.get("uuid")
+    if not uuid:
+        return render_template("login.html")
+
+    user = db.users.find_one({"uuid": uuid})
+
+    # If user is null or username does not matches with the username in the route.
+    if not user or user["name"] != username:
+        return render_template("login.html")
+
+    # Find package using package_name and namespace_name.
+    package = db.packages.find_one({"name": package_name, "namespace": namespace_name})
```

Comment:
as we can't parse _id by url thus, We would have to find the namespace `_id` by its name, as in packages collection we have namespace's `_id` .